utils/balance.ts

```diff
@@ -49,3 +52,24 @@ export function subscribeBalance(
     cb(availableBalance.toString())
   })
 }
+
+export const getNativeBalance = async ({
+  address,
+  api,
+  tokenId = 0,
+}: {
+  address: string
+  api: ApiPromise
+  tokenId: number
+}): Promise<string | number> => {
+  if (tokenId) {
+    return (
+      (await api.query.tokens.accounts(
+        address,
+        tokenId,
+      )) as PalletBalancesAccountData
+    ).free.toString()
+  }
+
+  return await balanceOf(api, address)
```

Comment:
mind you there is no need for `return await`
a simple `return` would return a promise the same

Reply:
indeed, only first one is needed